phylo2vec/src/tree_vec/ops/vector.rs

```diff
@@ -228,28 +229,26 @@ pub fn order_cherries_no_parents(ancestry: &mut Ancestry) {
     }
 }
 
-pub fn build_vector(cherries: Ancestry) -> Vec<usize> {
+pub fn build_vector(cherries: &Ancestry) -> Vec<usize> {
     let num_cherries = cherries.len();
-    let num_leaves = num_cherries + 1;
-
-    let mut v = vec![0; num_cherries];
-    let mut idxs = vec![0; num_leaves];
+    let mut v: Vec<usize> = vec![0; num_cherries];
 
     for i in 0..num_cherries {
         let [c1, c2, c_max] = cherries[i];
 
         let mut idx = 0;
 
-        for j in 1..c_max {
-            idx += idxs[j];
+        for j in 0..i {
+            if cherries[j][2] <= c_max {
+                idx += 1;
+            }
```

Comment:
While we've eliminated the need to have another array `idxs`, is this really more optimized? Need to evaluate.